sync/fork_test/fork.go

```diff
@@ -0,0 +1,155 @@
+package main
+
+import (
+	"context"
+	logging "github.com/ipfs/go-log/v2"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/go-header"
+	"github.com/celestiaorg/go-header/headertest"
+	"github.com/celestiaorg/go-header/local"
+	"github.com/celestiaorg/go-header/store"
+	"github.com/celestiaorg/go-header/sync"
+)
+
+// This program is for test purposes only. See TestForkFollowingPrevention
+// for further context.
+func main() {
+	t := &testing.T{}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	suite := headertest.NewTestSuite(t)
+	head := suite.Head()
+
+	maliciousSuite := headertest.NewTestSuiteWithHead(t, head)
+	maliciousHead := maliciousSuite.Head()
+
+	// set up syncer with a malicious peer as its remote peer
+	ee := newEclipsedExchange(ctx, t, head, maliciousHead)
+
+	localStore := store.NewTestStore(ctx, t, head)
+	syncer, err := sync.NewSyncer[*headertest.DummyHeader](
+		local.NewExchange[*headertest.DummyHeader](ee),
+		localStore,
+		headertest.NewDummySubscriber(),
+		// TrustingPeriod can be set to a nanosecond so even if the head
+		// given by the trusted peer expires by the time `subjectiveHead` is
+		// called again, it will still call Head on the `eclipsedExchange`
+		// which will return the same head as before.
+		sync.WithTrustingPeriod(time.Nanosecond),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	// give bad headers to the malicious (eclipsing) peer in order
+	// to attempt to get syncer to follow a fork
+	err = ee.appendToEclipsedExchange(ctx, maliciousSuite.GenDummyHeaders(99)...)
```

Comment:
Q: Is it giving bad headers to the eclipsed or to the eclipsing peer?

Reply:
It's appending a fork to the peer's store who will be "eclipsing" the syncer